pandas/tests/extension/integer/test_integer.py

```diff
@@ -567,6 +567,14 @@ def test_astype(self, all_data):
         expected = pd.Series(np.asarray(mixed))
         tm.assert_series_equal(result, expected)
 
+    def test_astype_nansafe(self):
+        # https://github.com/pandas-dev/pandas/pull/22343
+        arr = IntegerArray([np.nan, 1, 2], dtype="Int8")
+
+        with tm.assert_raises_regex(
+                ValueError, 'cannot convert float NaN to integer'):
+            arr.astype('uint32')
```

Comment:
Can you put this as a stand-alone test at the bottom of this file? (it's not overwriting a base test, but it is a custom integer-specific test)